lib/model.js

```diff
@@ -2858,27 +2859,41 @@ Model.create = async function create(doc, options) {
     return Array.isArray(doc) ? [] : null;
   }
   let res = [];
+  const immediateError = typeof options.immediateError === 'boolean' ? options.immediateError : true;
+
+  delete options.immediateError; // dont pass on the option to "$save"
+
   if (options.ordered) {
     for (let i = 0; i < args.length; i++) {
-      const doc = args[i];
-      const Model = this.discriminators && doc[discriminatorKey] != null ?
-        this.discriminators[doc[discriminatorKey]] || getDiscriminatorByValue(this.discriminators, doc[discriminatorKey]) :
-        this;
-      if (Model == null) {
-        throw new MongooseError(`Discriminator "${doc[discriminatorKey]}" not ` +
+      try {
+        const doc = args[i];
+        const Model = this.discriminators && doc[discriminatorKey] != null ?
+          this.discriminators[doc[discriminatorKey]] || getDiscriminatorByValue(this.discriminators, doc[discriminatorKey]) :
+          this;
+        if (Model == null) {
+          throw new MongooseError(`Discriminator "${doc[discriminatorKey]}" not ` +
           `found for model "${this.modelName}"`);
-      }
-      let toSave = doc;
-      if (!(toSave instanceof Model)) {
-        toSave = new Model(toSave);
-      }
+        }
+        let toSave = doc;
+        if (!(toSave instanceof Model)) {
+          toSave = new Model(toSave);
+        }
 
-      await toSave.$save(options);
-      res.push(toSave);
+        await toSave.$save(options);
+        res.push(toSave);
+      } catch (err) {
+        if (!immediateError) {
+          res.push(err);
+        } else {
+          throw err;
+        }
+      }
     }
     return res;
   } else {
-    res = await Promise.all(args.map(async doc => {
+    // ".bind(Promise)" is required, otherwise results in "TypeError: Promise.allSettled called on non-object"
+    const promiseType = !immediateError ? Promise.allSettled.bind(Promise) : Promise.all.bind(Promise);
+    let p = promiseType(args.map(async doc => {
```

Comment:
Why can't this be just `Promise.allSettled(args.map(...))` ? `bind()` is relatively slow, so I would prefer to avoid using that.

Reply:
i had avoided that because using `allSettled` in all cases would mean to both wait until all operations have finished, only to then throw the first error and second the `allSettled` result has to be mapped because of what it returns